web/src/api/signup/index.ts

```diff
@@ -0,0 +1,87 @@
+import { errorNotAllowed, errorResponse } from "src/backend/errors";
+import { NextApiRequest, NextApiResponse } from "next";
+import { setCookie } from "src/backend/cookies";
+import { getAPIServiceGraphqlClient } from "src/backend/graphql";
+import { generateUserJWT, verifySignUpJWT } from "src/backend/jwts";
+import * as yup from "yup";
+import { validateRequestSchema } from "src/backend/utils";
+import { getSdk } from "./graphql/signup.generated";
+import {
+  Session,
+  getSession,
+  updateSession,
+  withApiAuthRequired,
+} from "@auth0/nextjs-auth0";
+import { Auth0User } from "src/lib/types";
+import { isEmailUser } from "src/lib/utils";
+import { urls } from "src/lib/urls";
+
+export type SignupResponse = { returnTo: string };
+
+const schema = yup.object({
+  team_name: yup.string().strict().required(),
+  ironclad_id: yup.string().strict().required(),
+});
+
+export type SignupBody = yup.InferType<typeof schema>;
+
+export const handleSignup = withApiAuthRequired(
+  async (req: NextApiRequest, res: NextApiResponse<SignupResponse>) => {
+    if (!req.method || !["POST", "OPTIONS"].includes(req.method)) {
+      return errorNotAllowed(req.method, res, req);
+    }
+
+    const session = (await getSession(req, res)) as Session;
+    const auth0User = session?.user as Auth0User;
+
+    const { isValid, parsedParams, handleError } = await validateRequestSchema({
+      value: req.body,
+      schema,
+    });
+
+    if (!isValid || !parsedParams) {
+      return handleError(req, res);
+    }
+
+    const { team_name, ironclad_id } = parsedParams;
+
+    let nullifier_hash: string | undefined = undefined;
+
+    if (!isEmailUser(auth0User)) {
+      const nullifier = auth0User.sub.split("|")[2];
```

Comment:
let's not save new nullifier hashes from Auth0 for now

Reply:
What do you mean?
In the current implementation we need nullifier_hash in hasura to detect that user was registered with worldcoin on login-callback side.

Reply:
you're right, thought we were using different IDs